File: reactor-java/src/main/java/com/github/mostroverkhov/r2/reactor/internal/RequesterBuilder.java
package com.github.mostroverkhov.r2.reactor.internal;

import com.github.mostroverkhov.r2.core.CoreRequesterBuilder;
import com.github.mostroverkhov.r2.reactor.InteractionsInterceptor;
import com.github.mostroverkhov.r2.reactor.internal.adapters.RequesterAdapter;
import io.rsocket.RSocket;

import java.util.List;

public class RequesterBuilder extends CoreRequesterBuilder {

    public RequesterBuilder(RSocket rSocket, List<InteractionsInterceptor> interceptors) {
        adapter(new RequesterAdapter(rSocket, interceptors));
    }
}

File: rxjava-kotlin/src/main/java/com/github/mostroverkhov/r2/android/RequesterBuilder.kt
package com.github.mostroverkhov.r2.android

import com.github.mostroverkhov.r2.android.adapters.RequesterAdapter
import com.github.mostroverkhov.r2.core.CoreRequesterBuilder
import io.rsocket.android.RSocket

internal class RequesterBuilder(rSocket: RSocket) : CoreRequesterBuilder() {
    init {
        adapter(RequesterAdapter(rSocket))
    }
}